File: 1/2.py
with open('input.txt', 'r') as file:
    text = file.read()
    numbers = list(map(int, filter(lambda x: len(x) > 0, text.split('\n'))))
    for i in numbers:
        for j in numbers:
            for k in numbers:
                if i + j + k == 2020:
                    print(i * j * k)

File: 15/2.jl
numbers = parse.(Int, split(collect(eachline("input.txt"))[begin], ','))

const cache = Dict{Int, Int}()
turn = 0

for number in numbers
    global turn += 1
    cache[number] = turn
end

function next(number::Int)::Int
    global cache, turn
    turn += 1

    if turn == 30000001
        println(number)
        exit(0)
    end

    if number in keys(cache)
        n = turn - 1 - cache[number]
        cache[number] = turn - 1
        return n
    else
        cache[number] = turn - 1
        return 0
    end
end

last = numbers[end]

while true
    global last
    last = next(last)
end

